File: src/main/java/ni/edu/uam/michimaker_api/dto/TransformacionFeedDto.java
package ni.edu.uam.michimaker_api.dto;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class TransformacionFeedDto {

    private Integer id;

    private String nombreFiltro;
    private String fecha;

    // Compatibilidad con imágenes locales
    private String rutaImagen;

    // Nueva imagen almacenada en PostgreSQL
    private String imagenBase64;

    private Integer usuarioId;
    private String username;
    private String fotoPerfil;

    private String leyenda;

    public TransformacionFeedDto() {
    }
}

File: app/src/main/java/ni/edu/uam/michimaker/dto/TransformacionFeedDto.kt
package ni.edu.uam.michimaker.dto

data class TransformacionFeedDto(

    val id: Int,

    val nombreFiltro: String,

    val fecha: String,

    val rutaImagen: String?,

    val usuarioId: Int,

    val username: String,

    val fotoPerfil: String?,

    val leyenda: String?,

    val imagenBase64: String?
)